### PR TITLE
Fix scenario minimization

### LINKED ISSUE #215: Allow removing all operations in the parallel part during minimization

### PR DESCRIPTION
**Pull request by eupp**
Implement the following fixes in the scenario minimization:

* single-thread scenarios are not split into init/post parts, only parallel part with one thread is left;
* if after minimization parallel part of the first thread becomes empty, it is removed (the next non-empty thread in parallel part becomes first thread).

Closes #215.

### CODE REVIEW

**Comment by eupp** [comment]
I also add 2 new tests for minimization, that should emulate the problem reported in #207.

File: src/jvm/test/resources/expected_logs/validation_function_failure.txt [hunk 1]
```diff
@@ -23,8 +21,6 @@ The following interleaving leads to the error:
 | inc()    |
 | inc()    |
 | inc()    |
-| inc()    |
-| inc()    |
 | -------- |
 
 Detailed trace:
```
Comment (ndkoval) [inline src/jvm/test/resources/expected_logs/validation_function_failure.txt:26]: It's unclear from the output when the validation function fails

**Review by ndkoval — CHANGES_REQUESTED** [review]

**Review by ndkoval — APPROVED** [review]

File: src/jvm/test/resources/expected_logs/validation_function_failure.txt (hunk 1 above)
Comment (ndkoval) [reply src/jvm/test/resources/expected_logs/validation_function_failure.txt:26]: Sure, but can we do something with that? I suggest opening a separate issue and continuing the discussion there.

Comment (eupp) [reply src/jvm/test/resources/expected_logs/validation_function_failure.txt:26]: Ok, I opened the issue #220

**Pull request merged**